x-pack/plugins/cases/common/types/api/case/v1.ts

```diff
@@ -283,6 +283,21 @@ export const CasesFindRequestRt = rt.intersection([
   paginationSchema({ maxPerPage: MAX_CASES_PER_PAGE }),
 ]);
 
+export const CasesSearchRequestRt = rt.intersection([
+  rt.exact(
+    rt.partial({
+      /**
+       * custom fields of the case
+       */
+      customFields: rt.record(
+        rt.string,
+        rt.array(rt.union([rt.string, rt.boolean, rt.number, rt.null]))
```

Comment:
I understand we want to be flexible in supporting multi-select in the future but, is this an array so that we can

> Filter by multiple values of a custom field.

so we allow for searches like `customfield_id1: ['foobar', null]`?

And reading the original issue, I see the main approach is `The filtering relationship of the values should be logical OR`. How is this going to work together with the search for a multiple-selection custom field? if `customfield_id1` is a multiple-selection do we still apply the `OR` logic or when we search for these types we actually do `AND`?

PS: I am still reviewing the PR but I think there isn't any test where we send a request with multiple values, could you please add a couple of those @js-jankisalvi please 🙏

Reply:
yes, it is to allow search with multiple values and in case of boolean you can search with true, false and null options. 

`{
   "customFields": {
       "key_1": [null, true]
   }
}`
Above code will return cases with `"key_1"` custom field's value as `true or null`.
For multi-selection of same custom field it applies `OR logic`


 And if we have multiple custom fields then it uses `AND logic`
 `{
   "customFields": {
       "key_1": [null, true],
      "key_2": [true]
   }
}`
Above code gives cases with `"key_1": true or null `AND `"key_2" : true`

I have added multiple custom fields test in integration tests, but will also add it in `search.test`

Reply:
Maybe this is a problem for future us to solve so don't worry too much about it.

I was just thinking about the below

> For multi-selection of same custom field it applies OR logic

Does this make sense from a user perspective?  How are we going to allow users to search for cases that have multiple selection custom field X with option A AND option B selected? Because that kinda is the point of having multiple selection anyway right 😄